### ISSUE TITLE
getFitnessScore() possibe?

### ISSUE THREAD

**Issue by KastanDay** [STORM-IRIT/OpenGR#8]
Hello! Thank you for this work, it has sped up my alignment by an order of magnitude (30 sec --> 2 sec)!

I'm wondering if there is currently any way of retrieving the "Fitness score" which is printed upon alignment completion.  I see it is calculated and printed in the file `Super4PCS/include/pcl/registration/impl/super4pcs.hpp` but I do not see any easy way of extracting that score for my own use.  I would just like to ensure my fit is good before I declare my alignment complete. 

The line is:
``` C++
float score = matcher.ComputeTransformation(set1, &set2, final_transformation_, sampler, visitor);
```

If there is no easy way, maybe you could point me in the right direction for implementing it myself?

Thanks a million!
Kastan

**Comment by KastanDay** [comment]
Solution:
I added a public member variable `float fitness_score_;` to `Super4PCS/include/pcl/registration/super4pcs.h` then inside of `Super4PCS/include/pcl/registration/impl/super4pcs.hpp` I simply changed the above line to:
``` C++
  _fitness_score = matcher.ComputeTransformation(set1, &set2, final_transformation_, sampler, visitor);
```
Now in order to view the fitness score in my program I write:
``` C++
    pcl::Super4PCS<pcl::PointXYZ, pcl::PointXYZ> align;

    // set options (negative = feature disabled)
    align.options_.sample_size = _alignNumSamples;
    align.options_.max_time_seconds = _maxTimeSeconds;
    align.options_.delta = _alignDelta;
    align.setInputSource( _XYZobjectPointCloud );
    align.setInputTarget( XYZSceneCloud );

    std::cout << "===============================" << std::endl;
    {
        pcl::ScopeTime t("PCA Alignment");
        align.align ( *alignedObjectXYZCloud );
    }
    std::cout << "===============================" << std::endl;
    if (align.fitness_score_ > .8 ) { // publish alignment to robot }
```

**Comment by nmellado** [comment]
Hi @KastanDay ,

That's indeed a way to do it.
Please feel free to submit a PR with you changes (I'm out of work now and cannot update the code before a week, I'll do it later otherwise).
If you do submit a PR, I would prefer that `fitness_score_` stays protected and accessed using `float getFitnessScore() const`.

The other option is to give the possibility to set the `TransformationVisitor` from the wrapper. WIth this you can monitor the registration process and get the final score. It's a bit more advanced but also more powerful. Would you be interested by this ?

**Comment by KastanDay** [comment]
You are very correct, a protected member variable with a getter would be much better. I'd be happy to type that up and submit a PR. Full disclosure, as a undergrad developer this would be my first PR on an open source project - pretty exciting. 

I unfortunately can't figure out what you mean exactly by setting the `TransformationVisitor`.  I dug through the documentation and the PCL header files some, but I haven't found much. If you can point me in the right direction I'm always happy to do things the 'right' way, otherwise I'd gladly submit the simpler PR.

Cheers!

**Comment by nmellado** [comment]
Please fill free to submit the PR ;). Don't forget to add you in the contributor list by editing doc/overview.md

For the `TransformationVisitor`, it's ok I'll do it later (I keep the issues open in the meantime).

**Comment by nmellado** [comment]
Field added with 3b0ab9e608985e59d911deaaf37116f2bfc79fd6

**Comment by nmellado** [comment]
Fixed by #65
